Refactor/#4 authentication api

huchujj opened the pull request:
<!-- PR 작성 전에 우선 Reviewers, Assignees, label 지정하기 -->
## 🤨 Motivation
<!-- 코드를 추가/변경하게 된 이유 및 해결한 이슈 번호 
ex) - Resolved #2 -->
- Resolved #4 --> 토큰 재발급 API 변경에 따른 코드 수정

## 🔑 Key Changes 
<!-- 주요 구현 사항 -->
- **start_page.dart : _isValid 함수 수정**
refreshToken 유효 기간(30일) 만료 시 자동 로그인 해제
accessToken 유효 기간(1일) 만료 시 토큰 재발급 요청(getAccessToken)

- **apis.dart : getAccessToken 함수 수정**
responseData를 받아올 때, API 명세에 따라 코드 수정

## 🙏 To Reviewers 
<!-- 리뷰어에게 전달할 말 -->
- lib 폴더 구조를 변경하였습니다! 아래를 참고해주세요!
**config** : 프로젝트의 설정 파일 및 환경 변수를 저장하는 폴더입니다.
**mockdatas** : 테스트를 위해 사용하는 가짜 데이터(mock data)를 저장하는 폴더입니다.
**models** : 데이터 구조를 정의하는 클래스 및 객체 모델을 저장하는 폴더입니다.
**providers** : 상태 관리 및 데이터 공급자(provider) 클래스들을 저장하는 폴더입니다.
**repository** : 데이터 소스와 상호 작용하는 로직을 캡슐화하는 폴더입니다.
**util** : 유틸리티 함수 및 헬퍼 클래스들을 저장하는 폴더입니다.
**views** : UI 관련 코드(위젯, 화면 레이아웃 등)를 저장하는 폴더입니다
**main.dart** : 플러터 프로젝트의 집입점입니다.

huchujj committed:
refactor : 파일 및 폴더 구조 정리

issues : #4

huchujj committed:
chore : dart_jsonwebtoken 패키지 설치

issues : #4

huchujj committed:
refector : 토큰 재발급 api 수정

임시 커밋

issues : #4

huchujj committed:
refactor : 토큰 유효성 검사  _isValid 수정

refresh 토큰 유효기간 30일이 만료되면 자동 로그인 해제,
access 토큰 유효기간 1일(24시간)이 만료되면 getAccessToken 후 자동 로그인
음수 시간 오류 시 자동 로그인 해제

issues : #4

huchujj committed:
refactor : getAccessToken 함수 수정

getAccessToken 응답 받고 난 후의 코드, API 명세에 맞게 수정

issues : #4

mjj111 commented on lib/services/apis.dart:
```diff
@@ -512,9 +512,9 @@ class APIs {
-    print("accessToken : $accessToken");
-    print("refreshToken : $refreshToken");
-    print("타입 : ${accessToken.runtimeType},  ${refreshToken.runtimeType}");
+    // print("accessToken : $accessToken");
```
쓸모없어진 코드는 주석처리보다 지우는게 더 좋을 것 같습니다 ! 🤔

coke98 replied:
사실 디버깅용으로 남겨둔 것 같아서 전 별로 신경쓰이진 않는데,, 이슈로 등록해두고 최종적으로 지우거나, 디버깅 툴 사용하는 것도 방법도 좋을 수도 있겠네용

mjj111 commented on lib/views/pages/home/start_page.dart:
```diff
@@ -39,34 +40,46 @@ class _StartPageState extends State<StartPage> {
-      if (diff >= const Duration(days: 7)) {
-        //토큰 및 정보 지움
-        //자동로그인 해제
+      accessToken = await storage.read(key: 'token');
```
토큰 접근에 대한 로직을 다른 클래스나 메서드로 분리해서 호출하는 방식으로 수정하는 것은 어떨까요?!
start_page는 화면을 보이는데만 책임을 갖고,
다른 클래스 혹은 메서드는 토큰에 대한 접근 책임만을 갖는거죠!

coke98 replied:
메서드나 클래스를 분리하는 방법 중에 저는 개인적으로 
1. 역할
2. 행위의 추상화단계(추상적인 역할 - 보여준다, 구체적인 역할 - 해당 페이지에서 토큰에 관련된 일을 한다.)
를 기준으로 나누는 걸 선호해요

한 클래스, 메서드에서 로직이 길어진다면, 기준에 따라 (센스있고 알아보기 쉬운 네이밍의) 메서드나 클래스로 구분하면 더 읽기 쉬운 코드가 완성될 것 같아요 :)

mjj111 reviewed (approved):
고생하셨습니다! 리뷰 응답에 맞춰 수정하셔도 됩니다만..! 
제가 프론트 개발자가 아닌지라, 참고 정도로 받아주셨으면 감사하겠습니다  🙋‍♂️

mjj111 commented on lib/services/apis.dart:
```diff
@@ -505,16 +505,16 @@ class APIs {
 
 //토큰 재발급
```
api.dart 말고 
tokenApi
MemberApi
MatchingApi ...과 같이 API를 분리해서 관리하는건 어떨까요? 
API 파일이 굉장히 길어서 작업하기도 힘들 것 같습니다 🤔

coke98 replied:
분리 가능하면 하는것도 좋은 방법일지두

huchujj replied:
참고하겠습니다!🙂 확실히 더 유지보수하기도 쉬워질 것 같고, 가독성도 높아지겠네요~!!

huchujj committed:
refactor : 불필요한 print문 삭제

issues : #4

coke98 reviewed (approved):
첫 API 작업이군요! 첫 리뷰를 맡게되어 영광입니다...ㅎㅎ 앞으로도 잘 부탁드립니다 LGTM

huchujj committed:
refactor : start_page.dart에서 토큰 유효성 기능 분리

start_page의 isValid의 기능을 services/authenication_service.dart로 분리
토큰의 유효성에 대해 bool 형태의 값을 반환
start_page에서 반환 값에 따라 페이지 라우팅

issues : #4

huchujj committed:
refactor : AuthenticationService에서 TokenValidationService로 변경

issues : #4

huchujj committed:
refacor : apis.dart 분리

apis.dart를 api_service.dart, auth_service.dart, block_service.dart, board_service.dart, chat_service.dart, comment_service.dart, email_service.dart, market_service.dart, notification_service.dart, report_service.dart, user_service.dart로 분리

issues : #4
